client/src/components/History/ContentItem/DatasetCollection/DscUI.vue

```diff
@@ -6,12 +6,9 @@
         class="dataset dataset-collection collapsed"
         :class="{ selected }"
         :data-state="dsc.state"
-        @keydown.arrow-right.self.stop="$emit('select-collection', dsc)"
+        @keydown.arrow-right.self.stop="$emit('selectCollection')"
         @keydown.space.self.stop.prevent="$emit('update:selected', !selected)"
-        @click.stop="
-            $emit('select-collection', dsc);
-            $emit('update:expanded', dsc);
-        "
+        @click.stop="$emit('selectCollection')"
```

Comment:
What I'm trying to do is separate the UI elements from the data itself so that the behavior of the UI render-only elements can be more-easily re-used for different scenariosy.

In the situation you have highlighted I am simply emitting that this collection is selected and when the event is handled at a data-management layer, "selectCollection" is the event that is used to select a sub-collection to drill down into. That event is handled in History.vue.

https://github.com/Nerdinacan/galaxy/blob/1ebdb8fb00f47e7a754322b315558614d67b4c49/client/src/components/History/History.vue#L83

Reply:
Selecting a collection here is different from making a "content selection" for future operations.

The format of the event 'update:variable' was picked specifically to be compatible with standard Vue .sync syntax, i.e.

```html
<Component :var.sync="localVariable" />
```

Is the same as

```html
<Component :var="localVariable" @update:var="localVariable" />
```

But to answer your question, we are distinguishing between these three operations by handling their events in History.vue appropriately. Expansions are handled with a method exposed by the ExpandedItems provider, and selections will be handled by the SelectedItems Provider.

Reply:
Sure, that makes sense, but the way this is used in the history you're expanding this collection, which seems distinct from selecting a collection. How would you handle actual selection then (with checkboxes or filter or ...) ... shouldn't that emit `selectCollection` ? I don't mind either way, but this seems a little confusing to me, unless the idea is that you distinguish what the emitted event means in the parent component.

Reply:
```
<!-- from History.vue --->
<HistoryContentItem
        :item="item"
        :index="index"
        :show-selection="showSelection"
        
        <!-- here the value is being defined and its change is being trapped by the expanded provider -->
        :expanded="isExpanded(item)"
        @update:expanded="setExpanded(item, $event)"
    
        <!-- here is where selection is managed, I wanted this to happen in a separate data provider because we are about to expand on this concept to make it fancier, enabling more than just toggle-selections -->
        :selected="isSelected(item)"
        @update:selected="setSelected(item, $event)"
        @selectCollection="$emit('selectCollection', item)"
    />
```

The advantage of this setup is that most of the behavior of the history is handled in the History.vue comonent and the rest of the components are largely rendering.

Reply:
> (And whatever we do, we need to sync that with https://github.com/Nerdinacan/galaxy/blob/1ebdb8fb00f47e7a754322b315558614d67b4c49/client/src/components/WorkflowInvocationState/DatasetCollectionUIWrapper.vue#L9)

Actually you have highlighted the point of what I'm doing. By Making the event data-agnostic, you are free to manage those events how you please in your own component, as you actually have done. You are able to define your own application-appropriate wrapper for the renderer with it's own handler "toggleExpand" in your case.

Reply:
> Sure, that makes sense, but the way this is used in the history you're expanding this collection, which seems distinct from selecting a collection. How would you handle actual selection then (with checkboxes or filter or ...) ... shouldn't that emit `selectCollection` ? I don't mind either way, but this seems a little confusing to me, unless the idea is that you distinguish what the emitted event means in the parent component.

I think I'm having trouble interpreting your concern here. Is your concern "how do we select a collection if clicking on it drills down?" I think that's what you meant. Is it just a question of preferred UX?

Reply:
Yeah I just checked, i don't have any problems selecting a collection either via the space bar, or by clicking the explicit checkbox button in the toolbar. I believe this is similar to the current behavior. Perhaps I'm missing your point.

Reply:
I think github has some delays in there today. I'm just worried about the word `selectCollection`, since the way DscUI used in the history it seems like expanding is more appropriate ? But if you want something generic (select) and the parent component decides what this should actually do (expand or focus) then this is fine.

Reply:
Ahhh terminology! I see.

Yes:
1. Expanding opens a dataset to look at its details
2. Selecting adds content to the selection for a later operation
3. "SelectCollection" drills down into a collection

Yes I see your point that "SelectCollection" more closely resembles "expanding" than "selecting". I'll make some changes to make it more coherent. Good catch!

Reply:
I'm just going to call it viewCollection, so it's clearly a 3rd operation.